destination/destination.go

```diff
@@ -0,0 +1,167 @@
+	return sdk.DestinationWithMiddleware(&Destination{}, sdk.DefaultDestinationMiddleware()...)
+}
+
+// Parameters is a map of named Parameters that describe how to configure the Destination.
+func (d *Destination) Parameters() map[string]sdk.Parameter {
+	return map[string]sdk.Parameter{
+		config.KeyURI: {
+			Default:  "mongodb://localhost:27017",
+			Required: false,
+			Description: "The connection string. " +
+				"The URI can contain host names, IPv4/IPv6 literals, or an SRV record.",
+		},
+		config.KeyDB: {
+			Default:     "",
+			Required:    true,
+			Description: "The name of a database the connector must work with.",
+		},
+		config.KeyCollection: {
+			Default:     "",
+			Required:    true,
+			Description: "The name of a collection the connector must read from.",
+		},
+		config.KeyAuthUsername: {
+			Default:     "",
+			Required:    false,
+			Description: "The username.",
+		},
+		config.KeyAuthPassword: {
+			Default:     "",
+			Required:    false,
+			Description: "The user's password.",
+		},
+		config.KeyAuthDB: {
+			Default:     "admin",
+			Required:    false,
+			Description: "The name of a database that contains the user's authentication data.",
+		},
+		config.KeyAuthMechanism: {
+			Default:     "The default mechanism that defined depending on your MongoDB server version.",
+			Required:    false,
+			Description: "The authentication mechanism. ",
+		},
+		config.KeyAuthTLSCAFile: {
+			Default:  "",
+			Required: false,
+			Description: "The path to either a single or a bundle of certificate authorities" +
+				" to trust when making a TLS connection.",
+		},
+		config.KeyAuthTLSCertificateKeyFile: {
+			Default:     "",
+			Required:    false,
+			Description: "The path to the client certificate file or the client private key file.",
+		},
+	}
+}
+
+// Configure parses and initializes the config.
+func (d *Destination) Configure(ctx context.Context, cfg map[string]string) error {
+	configuration, err := config.Parse(cfg)
+	if err != nil {
+		return fmt.Errorf("parse config: %w", err)
+	}
+
+	d.config = configuration
+
+	return nil
+}
+
+// Open makes sure everything is prepared to receive records.
+func (d *Destination) Open(ctx context.Context) error {
+	var err error
+	d.client, err = mongo.Connect(ctx, d.config.GetClientOptions().SetRegistry(registry))
+	if err != nil {
+		return fmt.Errorf("connect to mongo: %w", err)
+	}
+
+	if err = d.client.Ping(ctx, nil); err != nil {
+		return fmt.Errorf("ping to mongo: %w", err)
+	}
+
+	collection, err := common.GetMongoCollection(ctx, d.client, d.config.DB, d.config.Collection)
+	if err != nil {
+		return fmt.Errorf("get mongo collection: %w", err)
+	}
+
+	d.writer = writer.NewWriter(collection)
+
+	return nil
+}
+
+// Write writes a record into a Destination.
+func (d *Destination) Write(ctx context.Context, records []sdk.Record) (int, error) {
+	for i, record := range records {
+		if err := d.writer.Write(ctx, record); err != nil {
```

Comment:
Hm.
`Write` writes each record differently depending on its `Operation`.
Or did you mean to collect records into slices by `Operation` and then write them in batches?
I think it will look complex.

Reply:
Yes, that's what I meant... I'll then create an issue for this so we look into it when appropriate.